agent.go

```diff
@@ -0,0 +1,155 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation.
+var _ Agents = (*agents)(nil)
+
+// Agents describes all the agent-related methods that the
+// Terraform Cloud API supports.
+// TFE API docs: https://www.terraform.io/docs/cloud/api/agents.html
+type Agents interface {
+	// Read an agent by its ID.
+	Read(ctx context.Context, agentID string) (*Agent, error)
+
+	// Read an agent by its ID with the given options.
+	ReadWithOptions(ctx context.Context, agentID string, options *AgentReadOptions) (*Agent, error)
+
+	// List all the agents of the given pool.
+	List(ctx context.Context, agentPoolID string, options *AgentListOptions) (*AgentList, error)
+}
+
+// agents implements Agents.
+type agents struct {
+	client *Client
+}
+
+// AgentList represents a list of agents.
+type AgentList struct {
+	*Pagination
+	Items []*Agent
+}
+
+// Agent represents a Terraform Cloud agent.
+type Agent struct {
+	ID   string `jsonapi:"primary,agents"`
+	Name string `jsonapi:"attr,name"`
+	IP   string `jsonapi:"attr,ip-address"`
+
+	// Relations
+	Organization *Organization `jsonapi:"relation,organization"`
+	Workspaces   []*Workspace  `jsonapi:"relation,workspaces"`
+}
+
+// A list of relations to include
+// https://www.terraform.io/cloud-docs/api-docs/agents#available-related-resources
+type AgentIncludeOpt string
+
+const (
+	AgentWorkspaces AgentIncludeOpt = "workspaces"
+)
+
+// AgentReadOptions represents the options for reading an agent.
+type AgentReadOptions struct {
+	Include []AgentIncludeOpt `url:"include,omitempty"`
+}
+
+// AgentListOptions represents the options for listing agents.
+type AgentListOptions struct {
+	ListOptions
+	// Optional: A list of relations to include. See available resources
+	// https://www.terraform.io/cloud-docs/api-docs/agents#available-related-resources
+	Include []AgentIncludeOpt `url:"include,omitempty"`
+}
+
+// Read a single agent by its ID
+func (s *agents) Read(ctx context.Context, agentID string) (*Agent, error) {
+	return s.ReadWithOptions(ctx, agentID, nil)
+}
+
+// Read a single agent by its ID with options.
+func (s *agents) ReadWithOptions(ctx context.Context, agentID string, options *AgentReadOptions) (*Agent, error) {
+	if !validStringID(&agentID) {
+		return nil, ErrInvalidAgentID //undeclared var name
+	}
+	if err := options.valid(); err != nil {
+		return nil, err
+	}
+
+	u := fmt.Sprintf("agents/%s", url.QueryEscape(agentID))
+	req, err := s.client.NewRequest("GET", u, nil)
```

Comment:
I think we should pass `options` here instead of `nil`.

Reply:
@sebasslash 👍 
@nfagerlund Can you elaborate on why it's a sign that the method, possibly, isn't needed❔